### ISSUE TITLE
Abort trap: 6 Assertion failed: (r == 0), function rwlock_wrlock, file tinycthread_extra.c

### ISSUE THREAD

**Issue by nijikon** [appsignal/rdkafka-ruby#95]
I run into this issue when running specs locally. I checked on `1.2.2-RC1`, `1.2.0` and `1.1.0`. Any ideas?

```
balrog:rdkafka-ruby n$ be rspec --profile 20 ./spec/rdkafka/consumer_spec.rb
.......F.......F.....FFFFFFF..F.....F....Assertion failed: (r == 0), function rwlock_wrlock, file tinycthread_extra.c, line 138.
Abort trap: 6
```

```
balrog:rdkafka-ruby n$ ruby -v
ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-darwin18]

balrog:rdkafka-ruby n$ brew config
HOMEBREW_VERSION: 2.1.15-107-g5da322d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5da322d6be314e5f8b1729e91e60be743a1e7d8e
Last commit: 31 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 28d42cb0f3160a0c8e169ce1a3aa30578d901c19
Core tap last commit: 26 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_AWS_ACCESS_KEY_ID: set
HOMEBREW_AWS_SECRET_ACCESS_KEY: set
HOMEBREW_BINTRAY_KEY: set
HOMEBREW_BINTRAY_USER: nijikon
HOMEBREW_DEVELOPER: 1
HOMEBREW_SANDBOX: 1
HOMEBREW_VERBOSE: 1
CPU: quad-core 64-bit haswell
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3/bin/ruby
Clang: 11.0 build 1100
Git: 2.23.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 13, 11.0.1, 1.8.0_45
macOS: 10.14.6-x86_64
CLT: 10.3.0.0.1.1562985497
Xcode: 11.2
CLT headers: 10.3.0.0.1.1562985497
XQuartz: 2.7.11 => /opt/X11
```

**Comment by nijikon** [comment]
Refs https://github.com/edenhill/librdkafka/issues/1000

**Comment by nijikon** [comment]
Refs https://github.com/appsignal/rdkafka-ruby/pull/94#issuecomment-549039807

**Comment by nijikon** [comment]
This is the spec that generates the issue. https://github.com/appsignal/rdkafka-ruby/blob/master/spec/rdkafka/consumer_spec.rb#L52-L93

**Comment by rgo** [comment]
Executing all specs in consumer_spec, I got:

```
Rdkafka::Consumer
  #subscripe, #unsubscribe and #subscription
    should subscribe, unsubscribe and return the subscription
    should raise an error when subscribing fails
    should raise an error when unsubscribing fails
    should raise an error when fetching the subscription fails
  #pause and #resume
    should raise when not TopicPartitionList
    should raise an error when pausing fails
    should raise an error when resume fails
    subscription
      should pause and then resume
  #seek
    should raise an error when seeking fails
    subscription
Assertion failed: (r == 0), function rwlock_wrlock, file tinycthread_extra.c, line 138.

[Process exited 134]
```

I added traces to locate the _assertion fail_ and went till: https://github.com/appsignal/rdkafka-ruby/blob/master/spec/rdkafka/consumer_spec.rb#L149 

Narrowing the search a little bit more I arrived to: consumer#poll https://github.com/appsignal/rdkafka-ruby/blob/master/lib/rdkafka/consumer.rb#L346

I'll take a deep look into it later.

**Comment by rgo** [comment]
Sadly, I'm stuck in consumer#poll. No (good) news :(

**Comment by thijsc** [comment]
I think there are some issues on the librdkafka side, but only on Mac. I've been pretty busy with other things, planning to get the dev and testing environment back to a good state again now.

**Comment by rgo** [comment]
At night, I'll try with [librdkafka 1.2.2](https://github.com/edenhill/librdkafka/releases/tag/v1.2.2) that was released yesterday.
I don't have big hopes because we were already using 1.2.2-rc1

**Comment by thijsc** [comment]
There's an upgrade to 1.2.2 in #98

**Comment by thijsc** [comment]
Are you still getting this when running #98?

**Comment by Adithya-copart** [comment]
I remember seeing this error while working on https://github.com/appsignal/rdkafka-ruby/pull/108 but don't recall the reason exactly.

**Comment by thijsc** [comment]
Are you still seeing this happen on master? A number of related fixes have been merged.

**Comment by thijsc** [comment]
This has probably been fixed, possibly also by #97. Feel free to reopen though.
